test/tests/misc/check_error/tests

```diff
@@ -640,23 +640,41 @@
   [../]
 
   [./unused_param_test]
-    type = 'RunApp'
-    expect_out = "unused parameter 'Mesh/unused_param'"
+    type = 'RunException'
+    expect_err = "unused parameter 'Mesh/unused_param'"
     input = 'unused_param_test.i'
-    allow_unused = true
 
-    requirement = 'The system shall report a warning when an unused parameter is provided through the input file or an Action.'
+    requirement = 'The system shall report an error when an unused parameter is provided through the input file or an Action.'
     issues = '#709 #759 #4101'
   [../]
 
   [./unused_param_test_cli]
-    type = 'RunApp'
-    expect_out = "unused parameter 'Executioner/unused_param'"
+    type = 'RunException'
+    expect_err = "unused parameter 'Executioner/unused_param'"
     input = 'unused_param_test.i'
     cli_args = 'Executioner/unused_param=true'
+
+    requirement = 'The system shall report an error when an unused parameter is supplied on the command line.'
+    issues = '#709 #759 #4101'
+  [../]
+
+  [./unused_param_test_warn]
+    type = 'RunApp'
+    expect_out = "unused parameter 'Mesh/unused_param'"
+    input = 'unused_param_test.i'
     allow_unused = true
 
-    requirement = 'The system shall report a warning when an unused parameter is supplied on the command line.'
+    requirement = 'The system shall report a warning when allow_unused = true has been set in the test spec.'
+    issues = '#709 #759 #4101'
+  [../]
+  
+  [./unused_param_test_warn_cli]
+    type = 'RunApp'
+    expect_out = "unused parameter 'Mesh/unused_param'"
+    input = 'unused_param_test.i'
+    cli_args = '--warn-unused'
+
+    requirement = 'The system shall report a warning when --warn-unused has been set in the cli_args'
```

Comment:
Requirements need to be more general and not include implementation details

```suggestion
    requirement = 'The system shall include the option to report a warning instead of an error for an unused parameter set in command line arguments'
```